### ISSUE TITLE
Get column names in output data

### ISSUE THREAD

**Issue by juba** [hyparam/hyparquet#24]
Hi,

I think this may be a very basic question but... when reading parquet data with `parquetRead`, I get an array of arrays like this:

```js
[[0, "foo"], [1, "bar"]]
```

I know that I can get the parquet column names from the parquet metadata with `parquetMetadata`. But is there a simple way to get data as an array of objects with column names, suitable to use with d3 or Plot?

Something like the following?

```js
[{column1: 0, column2: "foo"}, {column1: 1, column2: "bar"}]
```

Many thanks for your work on this project !

**Comment by platypii** [comment]
Honestly that's a good idea. I originally wrote this to dump row data into a table, so I just returned nested arrays. But I agree that it would be nicer in some cases to have it as objects.

I want to think about a couple things:

1. Is there a performance hit using objects instead of arrays? (or maybe a benefit?)
2. How to expose it without breaking the existing API? Probably need to come up with either a new function for reading, or perhaps an option to the existing `parquetRead` function.
3. Would require some changes to the code where hyparquet transposes from column-oriented data to row-oriented data [here](https://github.com/hyparam/hyparquet/blob/master/src/read.js#L187-L196)

This is something I will definitely consider adding as the API evolves. Pull requests are welcome if you are inclined to help!

**Comment by juba** [comment]
Thanks for your answer ! At least now I'm sure it was not something I missed from the API 😅

Thank you for taking my proposition into account. Even if the array-of-objects format is quite standard when using d3 / Plot / Observable, it may not be that useful outside of this ecosystem. 

Here is a very naive way to do the conversion from array of arrays to array of objects, but obviously that's not what could be used in hyparquet:

```js
const metadata = parquetMetadata(value.buffer);
const { children } = parquetSchema(metadata);
const columns = children.map((d) => d.element.name);
let table;
await parquetRead({
      file: value.buffer,
      onComplete: (data) => (table = data),
});
table = table.map((row) =>
    Object.fromEntries(row.map((d, i) => [columns[i], d]))
);
```

I'll see if I can dig a bit into hyparquet code and if my limited JavaScript skills could be enough to propose a PR.

Thanks !

**Comment by ctranstrum** [comment]
This issue is resolved with #25

**Comment by platypii** [comment]
Option available in published version 1.2.0 thanks @ctranstrum !

**Comment by juba** [comment]
Very nice, thanks a lot !
